### PR TITLE
Declaration count#2963

### PR DESCRIPTION
**Pull request by marinakr**

### CODE REVIEW

File: apps/core/lib/core/declaration_requests/api/approve.ex [hunk 1]
```diff
@@ -105,13 +108,26 @@ defmodule Core.DeclarationRequests.API.Approve do
...
     with %Employee{party: %Party{} = party} <- Employees.get_by_id(employee_id),
          employees <- Employees.get_active_by_party_id(party.id),
-         {:ok, %{"data" => %{"count" => declarations_count}}} <-
-           @ops_api.get_declarations_count(Enum.map(employees, &Map.get(&1, :id)), headers),
-         {:limit, true} <- {:limit, !party.declaration_limit || declarations_count < party.declaration_limit} do
+         employee_ids <- Enum.map(employees, &Map.get(&1, :id)),
+         declarations_request_count =
+           get_declarations_requests_count(DeclarationRequest.status(:approved), employee_ids),
+         {:ok, %{"data" => %{"count" => declarations_count}}} <- @ops_api.get_declarations_count(employee_ids, headers),
+         declarations_total_count = declarations_count + declarations_request_count,
```
Comment (AlexKovalevych) [inline apps/core/lib/core/declaration_requests/api/approve.ex]: same here

File: apps/core/lib/core/declaration_requests/api/approve.ex [hunk 2]
```diff
@@ -105,13 +108,26 @@ defmodule Core.DeclarationRequests.API.Approve do
   defp validate_declaration_limit(%DeclarationRequest{data: %{"employee" => %{"id" => employee_id}}}, headers) do
     with %Employee{party: %Party{} = party} <- Employees.get_by_id(employee_id),
          employees <- Employees.get_active_by_party_id(party.id),
-         {:ok, %{"data" => %{"count" => declarations_count}}} <-
-           @ops_api.get_declarations_count(Enum.map(employees, &Map.get(&1, :id)), headers),
-         {:limit, true} <- {:limit, !party.declaration_limit || declarations_count < party.declaration_limit} do
+         employee_ids <- Enum.map(employees, &Map.get(&1, :id)),
+         declarations_request_count =
```
Comment (AlexKovalevych) [inline apps/core/lib/core/declaration_requests/api/approve.ex]: please don't mix = and <- in the with statement, it's easy to miss it in the future and small change may break it (of course match everything in this case will always work)

File: apps/core/lib/core/declaration_requests/api/approve.ex [hunk 3]
```diff
@@ -105,13 +108,26 @@ defmodule Core.DeclarationRequests.API.Approve do
...
+         {:limit, true} <- {:limit, !party.declaration_limit || declarations_total_count < party.declaration_limit} do
       :ok
     else
       {:limit, false} -> Error.dump("This doctor reaches his limit and could not sign more declarations")
       _ -> {:error, {:conflict, "employee or party not found"}}
     end
   end
+
+  def get_declarations_requests_count(status, employee_ids) do
+    DeclarationRequest
```
Comment (AlexKovalevych) [inline apps/core/lib/core/declaration_requests/api/approve.ex]: will this query work fast enough?

Comment (marinakr) [reply apps/core/lib/core/declaration_requests/api/approve.ex]: checked few queries on prod, index by status and json data field used, works fast ~107ms

**Pull request merged**